### PR TITLE
Add NodeJS Trace integration test

### PR DESCRIPTION
**Pull request by geigerj**

### CODE REVIEW

File: generated/nodejs/test/trace_test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+var SPAN_KIND = traceV1.TraceSpan.SpanKind.RPC_CLIENT;
+
+var id = uuidjs.create().toString().replace(/-/g, '');
+var currentTime = Math.floor(new Date() / 1000);
+var startTime = {seconds: currentTime, nanos: 1234567}
+var endTime = {seconds: currentTime, nanos: 9876543}
+
+var api = traceV1.traceServiceApi();
+
+function test_patch(done) {
```
Comment (jmuk) [inline generated/nodejs/test/trace_test.js]: NodeJS prefers camelcase style for functions, so `testPatch`.

File: generated/nodejs/test/trace_test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    if (err) {
+      console.error(err);
+    }
+    console.log('patched trace');
+    // Wait a little for patched trace to propagate
+    setTimeout(done, 1000);
+  });
+}
+
+function test_get(done) {
```
Comment (jmuk) [inline generated/nodejs/test/trace_test.js]: ditto, `testGet`

File: generated/nodejs/test/trace_test.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if (err) {
+      console.error(err);
+    } else {
+      console.log('got trace:', trace);
+    }
+    done();
+  });
+}
+
+function test_list(done) {
```
Comment (jmuk) [inline generated/nodejs/test/trace_test.js]: ditto: `testList`

File: generated/nodejs/test/trace_test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+               spans: [span]};
+  var traces = {traces: [trace]};
+  console.log('test patch trace');
+  api.patchTraces(PROJECT_ID, traces, function(err) {
+    if (err) {
+      console.error(err);
+    }
+    console.log('patched trace');
+    // Wait a little for patched trace to propagate
+    setTimeout(done, 1000);
```
Comment (jmuk) [inline generated/nodejs/test/trace_test.js]: Is this true? Are there any documentation to justify the number?

Comment (geigerj) [inline generated/nodejs/test/trace_test.js]: Hmmm... I can anecdotally testify to the `get` not working without waiting ~1 sec (both in #78 and here) but I do not have documentation about what the expected latency after the `patchTraces` goes through to when it is available for `getTrace`.

File: generated/nodejs/test/trace_test.js (hunk 2 above)
Comment (geigerj) [inline generated/nodejs/test/trace_test.js]: Done

File: generated/nodejs/test/trace_test.js (hunk 3 above)
Comment (geigerj) [inline generated/nodejs/test/trace_test.js]: Done

File: generated/nodejs/test/trace_test.js (hunk 4 above)
Comment (jmuk) [inline generated/nodejs/test/trace_test.js]: That's fine then. I'm just trying to understand what's going on -- the current comment is good enough (I also double check with the proto file and it says nothing about that).

**Pull request merged**